gitdiff/patch_header.go

```diff
@@ -309,29 +322,39 @@ func scanMessageTitle(s *bufio.Scanner) (title string, indent string) {
 	return b.String(), indent
 }
 
-func scanMessageBody(s *bufio.Scanner, indent string) string {
-	var b strings.Builder
+func scanMessageBody(s *bufio.Scanner, indent string) (string, string) {
+	// Body and appendix
+	var body, appendix strings.Builder
+	c := &body
 	var empty int
 	for i := 0; s.Scan(); i++ {
 		line := s.Text()
-		if strings.TrimSpace(line) == "" {
+
+		line = strings.TrimRightFunc(line, unicode.IsSpace)
+		line = strings.TrimPrefix(line, indent)
+
+		if line == "" {
 			empty++
 			continue
 		}
 
-		if b.Len() > 0 {
-			b.WriteByte('\n')
+		// Parse out "appendix" information (often added by `git
+		// format-patch` and removed by `git am`).
+		if c == &body && line == "---" {
+			c = &appendix
```

Comment:
It might make sense to continue here, discarding the `---` line. While including it makes reassembling the original message easier, it seems like most uses of the `BodyAppendix` field would likely ignore or remove that first line.